microsoft-edge/webview2/how-to/webresourcerequested.md

```diff
@@ -0,0 +1,582 @@
+<!-- ## Example: Header modification when making a request -->
+
+<!-- this example doesn't exist in the sample repo -->
+
+<!-- note: the below intro is based on copying the main h2's Sentence 1 from above: -->
+In the following example, the host app _intercepts_ (receives) the document request that is sent from the WebView2 control to the `http://www.example.com` HTTP server, adds a custom header value and sends the request.  
+
+# [.NET](#tab/dotnet)
+
+```csharp
+// Add a filter to select all resource types under http://example.com
+webView.CoreWebView2.AddWebResourceRequestedFilter(
+      "http://www.example.com/*", CoreWebView2WebResourceContext.All);
+webView.CoreWebView2.WebResourceRequested += delegate (
+   object sender, CoreWebView2WebResourceRequestedEventArgs args) {
+   CoreWebView2WebResourceContext resourceContext = args.ResourceContext;
+   // Only intercept the document resources
+   if (resourceContext != CoreWebView2WebResourceContext.Document)
+   {
+      return;
+   }
+   CoreWebView2HttpRequestHeaders requestHeaders = args.Request.Headers;
+   requestHeaders.SetHeader("Custom", "Value");
+}
+```
+
+# [Win32](#tab/win32)
+
+```cpp
+// Add a filter to select all resource types under http://example.com
+m_webView->AddWebResourceRequestedFilter(
+      L"http://www.example.com/*", COREWEBVIEW2_WEB_RESOURCE_CONTEXT_ALL);
+m_webView->add_WebResourceRequested(
+      Callback<ICoreWebView2WebResourceRequestedEventHandler>(
+         [this](
+            ICoreWebView2* sender,
+            ICoreWebView2WebResourceRequestedEventArgs* args) {
+            COREWEBVIEW2_WEB_RESOURCE_CONTEXT resourceContext;
+            CHECK_FAILURE(args->get_ResourceContext(&resourceContext));
+            // Only intercept the document resources
+            if (resourceContext != COREWEBVIEW2_WEB_RESOURCE_CONTEXT_DOCUMENT)
+            {
+               return S_OK;
+            }
+
+            // Set a custom header to the document request.
+            wil::com_ptr<ICoreWebView2WebResourceRequest> request;
+            CHECK_FAILURE(args->get_Request(&request));
+            wil::com_ptr<ICoreWebView2HttpRequestHeaders> headers;
+            request->get_Headers(&headers);
+            headers->SetHeader(L"Custom", L"Value");
+            return S_OK;
+         })
+         .Get(),
+      &m_webResourceRequestedToken);
+```  
+
+---
+
+<!-- ====================================================================== -->
+## Overriding a response (to proactively replace it)
+<!-- Overriding and providing new responses to WebView2 -->
+<!-- ## Overriding the response and providing a different, custom response to the WebView2 control -->
+
+By default the HTTP server sends responses to the WebView2 control. Your host app can _override_ (ignore) a response that's sent from the HTTP server to the WebView2 control, and send a custom response to the WebView2 control instead of the original response.

```

Comment:
```suggestion
By default the HTTP server sends responses to the WebView2 control. Your host app can _override_ a response that's sent from the HTTP server to the WebView2 control, and send a custom response to the WebView2 control instead of the original response.

```